<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>microcolonizer analysis – Microcolonizers</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Microcolonizers</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./background.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">QIIME2 &amp; Temperature logger analysis</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./import-qc-metabarcoding.html"> 
<span class="menu-text">Preprocess &amp; compile data</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./analysis-microcolonizers.html" aria-current="page"> 
<span class="menu-text">Data analysis &amp; statistics</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#set-up-r-environment" id="toc-set-up-r-environment" class="nav-link active" data-scroll-target="#set-up-r-environment"><span class="header-section-number">1</span> Set up R environment</a>
  <ul class="collapse">
  <li><a href="#import-all-data" id="toc-import-all-data" class="nav-link" data-scroll-target="#import-all-data"><span class="header-section-number">1.1</span> Import all data</a></li>
  </ul></li>
  <li><a href="#subset-only-microcolonizer-data" id="toc-subset-only-microcolonizer-data" class="nav-link" data-scroll-target="#subset-only-microcolonizer-data"><span class="header-section-number">2</span> Subset only Microcolonizer data</a>
  <ul class="collapse">
  <li><a href="#plot-temperature-profiles-only" id="toc-plot-temperature-profiles-only" class="nav-link" data-scroll-target="#plot-temperature-profiles-only"><span class="header-section-number">2.1</span> Plot temperature profiles only</a></li>
  <li><a href="#join-temperature-and-tax-information" id="toc-join-temperature-and-tax-information" class="nav-link" data-scroll-target="#join-temperature-and-tax-information"><span class="header-section-number">2.2</span> Join temperature and tax information</a></li>
  <li><a href="#plot-species-richness-total-asvs-with-temperature" id="toc-plot-species-richness-total-asvs-with-temperature" class="nav-link" data-scroll-target="#plot-species-richness-total-asvs-with-temperature"><span class="header-section-number">2.3</span> Plot species richness (total ASVs) with temperature</a></li>
  </ul></li>
  <li><a href="#paring-down-the-data" id="toc-paring-down-the-data" class="nav-link" data-scroll-target="#paring-down-the-data"><span class="header-section-number">3</span> Paring down the data</a>
  <ul class="collapse">
  <li><a href="#eukaryotes" id="toc-eukaryotes" class="nav-link" data-scroll-target="#eukaryotes"><span class="header-section-number">3.1</span> Eukaryotes</a>
  <ul class="collapse">
  <li><a href="#filter-by-asv-distribution-18s" id="toc-filter-by-asv-distribution-18s" class="nav-link" data-scroll-target="#filter-by-asv-distribution-18s">Filter by ASV distribution (18S)</a></li>
  </ul></li>
  <li><a href="#bacteria" id="toc-bacteria" class="nav-link" data-scroll-target="#bacteria"><span class="header-section-number">3.2</span> Bacteria</a></li>
  <li><a href="#filter-for-network-analysis" id="toc-filter-for-network-analysis" class="nav-link" data-scroll-target="#filter-for-network-analysis"><span class="header-section-number">3.3</span> Filter for Network Analysis</a></li>
  </ul></li>
  <li><a href="#explore-research-questions-preliminary" id="toc-explore-research-questions-preliminary" class="nav-link" data-scroll-target="#explore-research-questions-preliminary"><span class="header-section-number">4</span> Explore research questions (preliminary)</a>
  <ul class="collapse">
  <li><a href="#presence-absense-with-barplots-to-id-taxa-distribution" id="toc-presence-absense-with-barplots-to-id-taxa-distribution" class="nav-link" data-scroll-target="#presence-absense-with-barplots-to-id-taxa-distribution"><span class="header-section-number">4.1</span> Presence absense with barplots to ID taxa distribution</a></li>
  <li><a href="#diversity-indices" id="toc-diversity-indices" class="nav-link" data-scroll-target="#diversity-indices"><span class="header-section-number">4.2</span> Diversity indices</a>
  <ul class="collapse">
  <li><a href="#s-diversity-indices" id="toc-s-diversity-indices" class="nav-link" data-scroll-target="#s-diversity-indices">18S diversity indices</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">microcolonizer analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="set-up-r-environment" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Set up R environment</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="import-all-data" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="import-all-data"><span class="header-section-number">1.1</span> Import all data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/GR_Microcolonizer_data.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  asv16s_df
  asv_wtax_18</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/temp-data.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  log_files_all</code></pre>
</div>
</div>
<p><code>asv_wtax_18</code> data includes 18S rRNA gene sequences from hydrothermal vents at Gorda Ridge and the microcolonizer substrates.</p>
<p><code>asv16s_df</code> same sample set as 18S, but with 16S rRNA gene sequences.</p>
</section>
</section>
<section id="subset-only-microcolonizer-data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Subset only Microcolonizer data</h1>
<p><em>GOALS</em></p>
<ol type="1">
<li><p>Visualize a summary of the protistan (18S rRNA gene) and bacteria/archaeal (16S rRNA gene) community within diffuse vents, at Mt Edwards especially, and then an overall look at the different microcolonizers.</p></li>
<li><p>Look at different microcolonizer samples with respect to the temperature profile. Are the communities different by temperature.</p></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(asv_wtax_18)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 29
  SAMPLE     FeatureID SEQUENCE_COUNT Taxon Domain Supergroup Phylum Class Order
  &lt;chr&gt;      &lt;chr&gt;              &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;
1 GordaRidg… 00056209…              8 Euka… Eukar… Stramenop… Sagen… &lt;NA&gt;  &lt;NA&gt; 
2 GordaRidg… 00056209…             13 Euka… Eukar… Stramenop… Sagen… &lt;NA&gt;  &lt;NA&gt; 
3 GordaRidg… 00096455…             91 Euka… Eukar… Rhizaria   Radio… Acan… &lt;NA&gt; 
4 GordaRidg… 00165708…              1 Euka… Eukar… Stramenop… Ochro… Pela… Pela…
5 GordaRidg… 0030ad8c…             36 Euka… Eukar… Stramenop… Opalo… MAST… MAST…
6 GordaRidg… 0030ad8c…             12 Euka… Eukar… Stramenop… Opalo… MAST… MAST…
# ℹ 20 more variables: Family &lt;chr&gt;, Genus &lt;chr&gt;, Species &lt;chr&gt;, VENT &lt;chr&gt;,
#   COORDINATES &lt;chr&gt;, SITE &lt;chr&gt;, SAMPLEID &lt;chr&gt;, SAMPLETYPE &lt;chr&gt;,
#   DEPTH &lt;chr&gt;, temp &lt;dbl&gt;, pH &lt;dbl&gt;, percseawater &lt;dbl&gt;, mg &lt;dbl&gt;, h2 &lt;chr&gt;,
#   h2s &lt;dbl&gt;, ch4 &lt;dbl&gt;, ProkConc &lt;chr&gt;, MC &lt;chr&gt;, dataset &lt;chr&gt;,
#   Substrate &lt;chr&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>mc_18s <span class="ot">&lt;-</span> asv_wtax_18 <span class="sc">|&gt;</span> </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SAMPLETYPE <span class="sc">==</span> <span class="st">"Microcolonizer"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(MC, Substrate, FeatureID, Taxon, Domain, </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>           Supergroup, Phylum, Class, Order, Family, Genus, Species) <span class="sc">|&gt;</span> </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SEQ_AVG =</span> <span class="fu">mean</span>(SEQUENCE_COUNT)) <span class="sc">|&gt;</span> </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(TAX_LEVEL, Supergroup, Phylum, <span class="at">sep =</span> <span class="st">"-"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'MC', 'Substrate', 'FeatureID', 'Taxon',
'Domain', 'Supergroup', 'Phylum', 'Class', 'Order', 'Family', 'Genus'. You can
override using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_18s)</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># head(asv16s_df)</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>mc_16s <span class="ot">&lt;-</span> asv16s_df <span class="sc">|&gt;</span> </span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sampletype <span class="sc">==</span> <span class="st">"Microcolonizer"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(MC, Substrate, FeatureID, Taxon, Domain, </span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>          Phylum, Class, Order, Family, Genus, Species) <span class="sc">|&gt;</span> </span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SEQ_AVG =</span> <span class="fu">mean</span>(SEQUENCE_COUNT)) <span class="sc">|&gt;</span> </span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(TAX_LEVEL, Domain, Phylum, <span class="at">sep =</span> <span class="st">"-"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'MC', 'Substrate', 'FeatureID', 'Taxon',
'Domain', 'Phylum', 'Class', 'Order', 'Family', 'Genus'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_16s)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="plot-temperature-profiles-only" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="plot-temperature-profiles-only"><span class="header-section-number">2.1</span> Plot temperature profiles only</h2>
<p>Get temperature</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(log_files_all)</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Factor by colors and pairs of MCs</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>mc_ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">6</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>log_files_all<span class="sc">$</span>MC_ORDER <span class="ot">&lt;-</span> <span class="fu">factor</span>(log_files_all<span class="sc">$</span>MC, <span class="at">levels =</span> mc_ids)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>mc_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#d7301f"</span>, <span class="st">"#4a1486"</span>, <span class="st">"#9e9ac8"</span>, <span class="st">"#fc8d59"</span>, <span class="st">"#2171b5"</span>, <span class="st">"#6baed6"</span>)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(mc_col) <span class="ot">&lt;-</span> mc_ids</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>temps <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(log_files_all, <span class="fu">aes</span>(<span class="at">x =</span> Parsed_time_EST, <span class="at">y =</span> Value, <span class="at">color =</span> <span class="fu">as.factor</span>(MC_ORDER))) <span class="sc">+</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_step</span>() <span class="sc">+</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mc_col) <span class="sc">+</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Temperature •C"</span>) <span class="sc">+</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>temps</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="analysis-microcolonizers_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="join-temperature-and-tax-information" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="join-temperature-and-tax-information"><span class="header-section-number">2.2</span> Join temperature and tax information</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>temp_row <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(log_files_all, <span class="fu">aes</span>(<span class="at">x =</span> Parsed_time_EST, <span class="at">y =</span> Value, <span class="at">color =</span> <span class="fu">as.factor</span>(MC_ORDER))) <span class="sc">+</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_step</span>() <span class="sc">+</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mc_col) <span class="sc">+</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(MC_ORDER)) <span class="sc">+</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Temperature •C"</span>) <span class="sc">+</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"left"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_16s)</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>bac_row <span class="ot">&lt;-</span> mc_16s <span class="sc">|&gt;</span> </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(TAX_LEVEL, MC, Substrate) <span class="sc">|&gt;</span> </span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SEQ_AVG)) <span class="sc">|&gt;</span> </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Substrate, <span class="at">y =</span> SUM, <span class="at">fill =</span> TAX_LEVEL)) <span class="sc">+</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(MC)) <span class="sc">+</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'TAX_LEVEL', 'MC'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>bac_row</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="analysis-microcolonizers_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_16s)</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>euk_row <span class="ot">&lt;-</span> mc_18s <span class="sc">|&gt;</span> </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(TAX_LEVEL, MC, Substrate) <span class="sc">|&gt;</span> </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SEQ_AVG)) <span class="sc">|&gt;</span> </span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Substrate, <span class="at">y =</span> SUM, <span class="at">fill =</span> TAX_LEVEL)) <span class="sc">+</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(MC)) <span class="sc">+</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'TAX_LEVEL', 'MC'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>euk_row</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="analysis-microcolonizers_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>temp_row <span class="sc">+</span> bac_row <span class="sc">+</span> euk_row <span class="sc">+</span> patchwork<span class="sc">::</span><span class="fu">plot_layout</span>(<span class="at">nrow =</span> <span class="dv">1</span>,<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.8</span>, <span class="fl">0.4</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="analysis-microcolonizers_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-species-richness-total-asvs-with-temperature" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="plot-species-richness-total-asvs-with-temperature"><span class="header-section-number">2.3</span> Plot species richness (total ASVs) with temperature</h2>
<p><em>Question</em>: Does overall species richness change within the microcolonizers? over “time” and with respect to temperature. Same for protists and prokaryotes?</p>
<p>Show differences between microcolonizers with temperature.</p>
<p><em>continuing question</em>: if there are specific patterns in species richness for certain microcolonizers, are there certain species that are driving this pattern?</p>
</section>
</section>
<section id="paring-down-the-data" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Paring down the data</h1>
<section id="eukaryotes" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="eukaryotes"><span class="header-section-number">3.1</span> Eukaryotes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(asv_wtax_18)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 29
  SAMPLE     FeatureID SEQUENCE_COUNT Taxon Domain Supergroup Phylum Class Order
  &lt;chr&gt;      &lt;chr&gt;              &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;
1 GordaRidg… 00056209…              8 Euka… Eukar… Stramenop… Sagen… &lt;NA&gt;  &lt;NA&gt; 
2 GordaRidg… 00056209…             13 Euka… Eukar… Stramenop… Sagen… &lt;NA&gt;  &lt;NA&gt; 
3 GordaRidg… 00096455…             91 Euka… Eukar… Rhizaria   Radio… Acan… &lt;NA&gt; 
4 GordaRidg… 00165708…              1 Euka… Eukar… Stramenop… Ochro… Pela… Pela…
5 GordaRidg… 0030ad8c…             36 Euka… Eukar… Stramenop… Opalo… MAST… MAST…
6 GordaRidg… 0030ad8c…             12 Euka… Eukar… Stramenop… Opalo… MAST… MAST…
# ℹ 20 more variables: Family &lt;chr&gt;, Genus &lt;chr&gt;, Species &lt;chr&gt;, VENT &lt;chr&gt;,
#   COORDINATES &lt;chr&gt;, SITE &lt;chr&gt;, SAMPLEID &lt;chr&gt;, SAMPLETYPE &lt;chr&gt;,
#   DEPTH &lt;chr&gt;, temp &lt;dbl&gt;, pH &lt;dbl&gt;, percseawater &lt;dbl&gt;, mg &lt;dbl&gt;, h2 &lt;chr&gt;,
#   h2s &lt;dbl&gt;, ch4 &lt;dbl&gt;, ProkConc &lt;chr&gt;, MC &lt;chr&gt;, dataset &lt;chr&gt;,
#   Substrate &lt;chr&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(asv_wtax_18<span class="sc">$</span>FeatureID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5886</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(asv_wtax_18<span class="sc">$</span>Domain)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Eukaryota"  "Bacteria"   "Unassigned"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>mc_18S_df <span class="ot">&lt;-</span> asv_wtax_18 <span class="sc">|&gt;</span> </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>SAMPLE) <span class="sc">|&gt;</span> </span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset to data for THIS specific project</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SAMPLETYPE <span class="sc">==</span> <span class="st">"Microcolonizer"</span> <span class="sc">|</span> VENT <span class="sc">==</span> <span class="st">"Mt Edwards"</span> <span class="sc">|</span> VENT <span class="sc">==</span> <span class="st">"Deep seawater"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">|&gt;</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUBSTRATE =</span> <span class="fu">str_replace_all</span>(Substrate, <span class="st">"z2"</span>, <span class="st">"z"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE =</span> <span class="fu">case_when</span>(</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>    SAMPLETYPE <span class="sc">==</span> <span class="st">"Microcolonizer"</span> <span class="sc">~</span> <span class="fu">paste</span>(MC, SUBSTRATE, <span class="at">sep =</span> <span class="st">"-"</span>),</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>    VENT <span class="sc">==</span> <span class="st">"Deep seawater"</span> <span class="sc">~</span> <span class="st">"Background seawater"</span>,</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>    VENT <span class="sc">==</span> <span class="st">"Mt Edwards"</span> <span class="sc">~</span> <span class="st">"Mt Edwards diffuse fluid"</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">|&gt;</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, FeatureID, Taxon, Domain, </span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>           Supergroup, Phylum, Class, Order, Family, Genus, Species) <span class="sc">|&gt;</span> </span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SEQ_AVG =</span> <span class="fu">mean</span>(SEQUENCE_COUNT)) <span class="sc">|&gt;</span> </span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SEQ_AVG <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE', 'FeatureID', 'Taxon', 'Domain',
'Supergroup', 'Phylum', 'Class', 'Order', 'Family', 'Genus'. You can override
using the `.groups` argument.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_tmp_18)</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(mc_tmp_18$Domain)</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(mc_tmp_18$SAMPLE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="filter-by-asv-distribution-18s" class="level3">
<h3 class="anchored" data-anchor-id="filter-by-asv-distribution-18s">Filter by ASV distribution (18S)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ASV filtering by how many sequences are present for each ASV</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Total number of ASVs with Unassigned</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(mc_18S_df<span class="sc">$</span>FeatureID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2168</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ID substrates only</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>mc_tmp_18_onsubstrates <span class="ot">&lt;-</span> mc_18S_df <span class="sc">%&gt;%</span> </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"MC"</span>, SAMPLE))</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(mc_tmp_18_onsubstrates<span class="sc">$</span>SAMPLE) <span class="co"># check</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "MC1-Quartz" "MC1-Riftia" "MC1-Shell"  "MC2-Quartz" "MC2-Riftia"
 [6] "MC3-Riftia" "MC3-Shell"  "MC4-Quartz" "MC4-Shell"  "MC5-Quartz"
[11] "MC5-Riftia" "MC5-Shell"  "MC6-Quartz" "MC6-Riftia" "MC6-Shell" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(mc_tmp_18_onsubstrates<span class="sc">$</span>FeatureID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1365</code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Repeat for opposite:</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>bkgd_vent <span class="ot">&lt;-</span> mc_18S_df <span class="sc">%&gt;%</span> </span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">grepl</span>(<span class="st">"MC"</span>, SAMPLE)))</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(bkgd_vent<span class="sc">$</span>SAMPLE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Background seawater"      "Mt Edwards diffuse fluid"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Isolate a list of feature IDs in background and vent fluid</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>bkgd_vent_ASV_list <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(bkgd_vent<span class="sc">$</span>FeatureID))</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co"># To get substrate ONLY, remove this character list</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>mc_tmp_18_onsubstrates_ONLY <span class="ot">&lt;-</span> mc_tmp_18_onsubstrates <span class="sc">%&gt;%</span> </span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(FeatureID <span class="sc">%in%</span> bkgd_vent_ASV_list))</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mc_tmp_18_onsubstrates_ONLY)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 12
# Groups:   SAMPLE, FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order,
#   Family, Genus [6]
  SAMPLE     FeatureID   Taxon Domain Supergroup Phylum Class Order Family Genus
  &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;
1 MC1-Quartz 00b72d1a5f… Euka… Eukar… Alveolata  Cilio… Spir… Eupl… Aspid… Aspi…
2 MC1-Quartz 041415f220… Euka… Eukar… Alveolata  Cilio… Phyl… Suct… Sucto… Suct…
3 MC1-Quartz 078cb4d468… Euka… Eukar… Rhizaria   Cerco… Endo… Vamp… Vampy… &lt;NA&gt; 
4 MC1-Quartz 09b4192e90… Euka… Eukar… Hacrobia   Picoz… Pico… Pico… Picoz… Pico…
5 MC1-Quartz 0de01cdd2b… Euka… Eukar… Stramenop… Opalo… &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt; 
6 MC1-Quartz 0e89357d45… Euka… Eukar… Alveolata  Dinof… &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt; 
# ℹ 2 more variables: Species &lt;chr&gt;, SEQ_AVG &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(mc_tmp_18_onsubstrates_ONLY<span class="sc">$</span>FeatureID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 999</code></pre>
</div>
</div>
</section>
</section>
<section id="bacteria" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="bacteria"><span class="header-section-number">3.2</span> Bacteria</h2>
<p>As long as an ASV appears at least once to have more than 100 sequences, keep the ASV.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(asv16s_df)</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>mc_16S_df <span class="ot">&lt;-</span> asv16s_df <span class="sc">|&gt;</span> </span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sampletype <span class="sc">==</span> <span class="st">"Microcolonizer"</span> <span class="sc">|</span> LocationName <span class="sc">==</span> <span class="st">"Mt Edwards Vent"</span> <span class="sc">|</span> LocationName <span class="sc">==</span> <span class="st">"Deep seawater"</span> <span class="sc">&amp;</span> STATUS <span class="sc">==</span> <span class="st">"keep"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE =</span> <span class="fu">case_when</span>(</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>    Sampletype <span class="sc">==</span> <span class="st">"Microcolonizer"</span> <span class="sc">~</span> <span class="fu">paste</span>(MC, Substrate, <span class="at">sep =</span> <span class="st">"-"</span>),</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>    LocationName <span class="sc">==</span> <span class="st">"Deep seawater"</span> <span class="sc">~</span> <span class="st">"Background seawater"</span>,</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>    LocationName <span class="sc">==</span> <span class="st">"Mt Edwards Vent"</span> <span class="sc">~</span> <span class="st">"Mt Edwards diffuse fluid"</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">|&gt;</span> </span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, FeatureID, Taxon, Domain, </span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>          Phylum, Class, Order, Family, Genus, Species) <span class="sc">|&gt;</span> </span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SEQ_AVG =</span> <span class="fu">mean</span>(SEQUENCE_COUNT)) <span class="sc">|&gt;</span> </span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(Domain <span class="sc">==</span> <span class="st">"Unassigned"</span>) <span class="sc">&amp;</span> <span class="sc">!</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SEQ_AVG <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE', 'FeatureID', 'Taxon', 'Domain',
'Phylum', 'Class', 'Order', 'Family', 'Genus'. You can override using the
`.groups` argument.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mc_18S_df, mc_16S_df, <span class="at">file =</span> <span class="st">"input-data/microcolonizer_inputdata_02082025.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>ML work on ensuring data files and tables regarding ASV distribution are determined.</p>
</blockquote>
</section>
<section id="filter-for-network-analysis" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="filter-for-network-analysis"><span class="header-section-number">3.3</span> Filter for Network Analysis</h2>
<blockquote class="blockquote">
<p>SH</p>
</blockquote>
<p>Create filtered 18S dataset and convert to phyloseq object.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_tmp_18)</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="co"># mat_18s_filtered &lt;- mc_tmp_18 |&gt; ungroup() |&gt; </span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(FeatureID %in% keep_18) |&gt; </span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(SEQ_AVG &gt; 0) |&gt; </span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   select(FeatureID, SAMPLE, SEQ_AVG) |&gt; </span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   pivot_wider(names_from = SAMPLE, values_from = SEQ_AVG, values_fill = 0) |&gt; </span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   select(-`MC4-Shell`) |&gt; </span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   column_to_rownames(var = "FeatureID") |&gt; </span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   as.matrix()</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a><span class="co"># # colnames(mat_18s_filtered)</span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a><span class="co"># mat_tax18_filtered &lt;- mc_tmp_18 |&gt; ungroup() |&gt; </span></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(FeatureID %in% keep_18) |&gt; </span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   select(FeatureID, Taxon, Domain, Phylum, Class, Order, Family, Genus, Species) |&gt; </span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   distinct() |&gt; </span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a><span class="co">#   column_to_rownames(var = "FeatureID") |&gt; </span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a><span class="co">#   as.matrix()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>SPEIC-EASI network analysis requires samples to be in the same order. Additionally, the bacteria set of samples has more substrate options than the eukaryote dataset.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(mat_18s_filtered)</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set_of_samples &lt;- colnames(mat_18s_filtered) # Make character list</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="co"># class(set_of_samples)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set_of_samples</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(tmptmp$SAMPLE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Create filtered 16S dataset and convert to phyloseq object.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_tmp)</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="co"># mat_16s_filtered &lt;- mc_tmp |&gt; ungroup() |&gt; </span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(FeatureID %in% keep) |&gt; </span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(SEQ_AVG &gt; 0) |&gt; </span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   select(FeatureID, SAMPLE, SEQ_AVG) |&gt; </span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   pivot_wider(names_from = SAMPLE, values_from = SEQ_AVG, values_fill = 0) |&gt; </span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   ungroup() |&gt; </span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   select(FeatureID, all_of(set_of_samples)) |&gt; </span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   column_to_rownames(var = "FeatureID") |&gt; </span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   as.matrix()</span></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(mat_16s_filtered)</span></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a><span class="co"># mat_tax_filtered &lt;- mc_tmp |&gt; ungroup() |&gt; </span></span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(FeatureID %in% keep) |&gt; </span></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   select(FeatureID, Taxon, Domain, Phylum, Class, Order, Family, Genus, Species) |&gt; </span></span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a><span class="co">#   distinct() |&gt; </span></span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a><span class="co">#   column_to_rownames(var = "FeatureID") |&gt; </span></span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   as.matrix()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># identical(colnames(mat_16s_filtered), colnames(mat_18s_filtered))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="explore-research-questions-preliminary" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Explore research questions (preliminary)</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/microcolonizer_inputdata_02082025.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  mc_18S_df
  mc_16S_df</code></pre>
</div>
</div>
<section id="presence-absense-with-barplots-to-id-taxa-distribution" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="presence-absense-with-barplots-to-id-taxa-distribution"><span class="header-section-number">4.1</span> Presence absense with barplots to ID taxa distribution</h2>
<ul>
<li>incorporate ML’s code to here!</li>
<li>maybe remove low relative abundance ASVs?</li>
</ul>
</section>
<section id="diversity-indices" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="diversity-indices"><span class="header-section-number">4.2</span> Diversity indices</h2>
<section id="s-diversity-indices" class="level3">
<h3 class="anchored" data-anchor-id="s-diversity-indices">18S diversity indices</h3>
<p>Prep 2 different data types:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mc_18S_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 12
# Groups:   SAMPLE, FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order,
#   Family, Genus [6]
  SAMPLE       FeatureID Taxon Domain Supergroup Phylum Class Order Family Genus
  &lt;chr&gt;        &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;
1 Background … 00165708… Euka… Eukar… Stramenop… Ochro… Pela… Pela… Pelag… Pela…
2 Background … 0038478b… Euka… Eukar… Opisthoko… Metaz… Cnid… Cnid… Hydro… &lt;NA&gt; 
3 Background … 0132268c… Euka… Eukar… Alveolata  Cilio… Spir… &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt; 
4 Background … 01a71ee7… Euka… Eukar… Stramenop… Pseud… MAST… MAST… MAST-… MAST…
5 Background … 01d1a4a1… Euka… Eukar… Rhizaria   Radio… RAD-B RAD-… RAD-B… RAD-…
6 Background … 02029510… Euka… Eukar… Rhizaria   Radio… RAD-B RAD-… RAD-B… RAD-…
# ℹ 2 more variables: Species &lt;chr&gt;, SEQ_AVG &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(mc_18S_df<span class="sc">$</span>SAMPLE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Background seawater"      "MC1-Quartz"              
 [3] "MC1-Riftia"               "MC1-Shell"               
 [5] "MC2-Quartz"               "MC2-Riftia"              
 [7] "MC3-Riftia"               "MC3-Shell"               
 [9] "MC4-Quartz"               "MC4-Shell"               
[11] "MC5-Quartz"               "MC5-Riftia"              
[13] "MC5-Shell"                "MC6-Quartz"              
[15] "MC6-Riftia"               "MC6-Shell"               
[17] "Mt Edwards diffuse fluid"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>sum_across_substrates <span class="ot">&lt;-</span> mc_18S_df <span class="sc">%&gt;%</span> </span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(SAMPLE, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"SAMPLE_ID"</span>, <span class="st">"SUBSTRATE"</span>), <span class="at">sep =</span> <span class="st">"-"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE_ID, FeatureID) <span class="sc">%&gt;%</span> </span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SEQ_AVG))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 2 pieces. Missing pieces filled with `NA` in 1378 rows [1, 2, 3, 4, 5,
6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE_ID'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>microcol_only <span class="ot">&lt;-</span> mc_18S_df <span class="sc">%&gt;%</span> </span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(SAMPLE, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"SAMPLE_ID"</span>, <span class="st">"SUBSTRATE"</span>), <span class="at">sep =</span> <span class="st">"-"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(SUBSTRATE)) <span class="sc">%&gt;%</span> </span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, SUBSTRATE, FeatureID) <span class="sc">%&gt;%</span> </span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SEQ_AVG))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 2 pieces. Missing pieces filled with `NA` in 1378 rows [1, 2, 3, 4, 5,
6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE', 'SUBSTRATE'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(sum_across_substrates$SUBSTRATE)</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="co"># head(microcol_only)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot total ASVs on y-axis for each sample (x-axis).</p>
<p>Plot Shannon diversity on y-axis for each sample (x-axis).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First make dataset wide with columns as samples and featureIDs as rownames</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>wide_18S <span class="ot">&lt;-</span> mc_18S_df <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select only columns we need</span></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SAMPLE, FeatureID, SEQ_AVG) <span class="sc">%&gt;%</span> </span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> SAMPLE, <span class="at">values_from =</span> SEQ_AVG, <span class="at">values_fill =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"FeatureID"</span>)</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: permute</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: lattice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>This is vegan 2.6-8</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>?<span class="fu">vegdist</span>()</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>?<span class="fu">diversity</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Help on topic 'diversity' was found in the following packages:

  Package               Library
  igraph                /Users/skhu/Library/R/arm64/4.4/library
  vegan                 /Users/skhu/Library/R/arm64/4.4/library


Using the first match ...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(wide_18S)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "data.frame"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>wide_18S_mat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(wide_18S)</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Standardize data ahead of time</span></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a><span class="co"># CLR, hellinger, standardize</span></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>stand_wide_18S_mat <span class="ot">&lt;-</span> <span class="fu">decostand</span>(wide_18S_mat, <span class="at">method =</span> <span class="st">"hellinger"</span>, <span class="at">MARGIN =</span> <span class="dv">2</span>, )</span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(stand_wide_18S_mat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                 Background seawater  MC1-Quartz MC1-Riftia
00165708dab37d750eb573b1720ff8bd         0.006425456 0.000000000 0.00000000
0038478be7fb4f097ce93a5e9341af2a         0.030815405 0.000000000 0.00000000
0132268c3014d4afd36f25e499f05fb6         0.011129217 0.000000000 0.00000000
01a71ee728b1597b04d339505a58ed04         0.039609172 0.032305762 0.04812114
01d1a4a17e3a26ee76b34b62cb0cbef8         0.019276368 0.000000000 0.00000000
020295103ca8304135054e04d9110899         0.017000159 0.005347288 0.01584049
                                 MC1-Shell  MC2-Quartz MC2-Riftia MC3-Riftia
00165708dab37d750eb573b1720ff8bd         0 0.000000000          0 0.00000000
0038478be7fb4f097ce93a5e9341af2a         0 0.000000000          0 0.00000000
0132268c3014d4afd36f25e499f05fb6         0 0.000000000          0 0.00000000
01a71ee728b1597b04d339505a58ed04         0 0.009075374          0 0.01542814
01d1a4a17e3a26ee76b34b62cb0cbef8         0 0.000000000          0 0.00000000
020295103ca8304135054e04d9110899         0 0.000000000          0 0.00000000
                                 MC3-Shell MC4-Quartz MC4-Shell  MC5-Quartz
00165708dab37d750eb573b1720ff8bd         0  0.0000000         0 0.000000000
0038478be7fb4f097ce93a5e9341af2a         0  0.0000000         0 0.000000000
0132268c3014d4afd36f25e499f05fb6         0  0.0000000         0 0.000000000
01a71ee728b1597b04d339505a58ed04         0  0.0116829         0 0.017606704
01d1a4a17e3a26ee76b34b62cb0cbef8         0  0.0000000         0 0.004496347
020295103ca8304135054e04d9110899         0  0.0000000         0 0.000000000
                                 MC5-Riftia  MC5-Shell  MC6-Quartz MC6-Riftia
00165708dab37d750eb573b1720ff8bd 0.00000000 0.00000000 0.000000000  0.0000000
0038478be7fb4f097ce93a5e9341af2a 0.00000000 0.00000000 0.000000000  0.0000000
0132268c3014d4afd36f25e499f05fb6 0.00000000 0.00000000 0.000000000  0.0000000
01a71ee728b1597b04d339505a58ed04 0.01145139 0.00778983 0.023881258  0.0113429
01d1a4a17e3a26ee76b34b62cb0cbef8 0.00000000 0.00000000 0.007725547  0.0000000
020295103ca8304135054e04d9110899 0.00000000 0.00000000 0.000000000  0.0000000
                                 MC6-Shell Mt Edwards diffuse fluid
00165708dab37d750eb573b1720ff8bd         0               0.00000000
0038478be7fb4f097ce93a5e9341af2a         0               0.00000000
0132268c3014d4afd36f25e499f05fb6         0               0.00000000
01a71ee728b1597b04d339505a58ed04         0               0.07891899
01d1a4a17e3a26ee76b34b62cb0cbef8         0               0.01884863
020295103ca8304135054e04d9110899         0               0.01642385</code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shannon diversity:</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>shannon_18S <span class="ot">&lt;-</span> <span class="fu">diversity</span>(stand_wide_18S_mat, <span class="at">index =</span> <span class="st">"shannon"</span>, <span class="at">MARGIN =</span> <span class="dv">2</span>, <span class="at">equalize.groups =</span> <span class="cn">TRUE</span>)</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>shannon_18S</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     Background seawater               MC1-Quartz               MC1-Riftia 
                5.926298                 5.239540                 5.265270 
               MC1-Shell               MC2-Quartz               MC2-Riftia 
                3.572935                 3.133511                 2.728965 
              MC3-Riftia                MC3-Shell               MC4-Quartz 
                3.212243                 2.943589                 4.327854 
               MC4-Shell               MC5-Quartz               MC5-Riftia 
                3.771707                 4.715165                 4.579419 
               MC5-Shell               MC6-Quartz               MC6-Riftia 
                4.446500                 5.044743                 4.912915 
               MC6-Shell Mt Edwards diffuse fluid 
                3.709465                 6.465990 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Inv Simp</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>inv_simp_18S <span class="ot">&lt;-</span> <span class="fu">diversity</span>(stand_wide_18S_mat, <span class="at">index =</span> <span class="st">"invsimpson"</span>, <span class="at">MARGIN =</span> <span class="dv">2</span> , <span class="at">equalize.groups =</span> <span class="cn">TRUE</span>)</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>inv_simp_18S</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     Background seawater               MC1-Quartz               MC1-Riftia 
              289.628959                69.508886                62.863905 
               MC1-Shell               MC2-Quartz               MC2-Riftia 
               26.838914                10.009635                11.239714 
              MC3-Riftia                MC3-Shell               MC4-Quartz 
                9.543133                14.485855                37.263324 
               MC4-Shell               MC5-Quartz               MC5-Riftia 
               28.701510                43.423979                44.809558 
               MC5-Shell               MC6-Quartz               MC6-Riftia 
               51.502536                68.066439                63.124399 
               MC6-Shell Mt Edwards diffuse fluid 
               31.444057               381.750826 </code></pre>
</div>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>