<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Microcolonizers - microcolonizer analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Microcolonizers</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./background.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">QIIME2 &amp; Temperature logger analysis</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./import-qc-metabarcoding.html" rel="" target="">
 <span class="menu-text">Preprocess &amp; compile data</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./analysis-microcolonizers.html" rel="" target="" aria-current="page">
 <span class="menu-text">Data analysis &amp; statistics</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#set-up-r-environment" id="toc-set-up-r-environment" class="nav-link active" data-scroll-target="#set-up-r-environment"><span class="header-section-number">1</span> Set up R environment</a>
  <ul class="collapse">
  <li><a href="#import-all-data" id="toc-import-all-data" class="nav-link" data-scroll-target="#import-all-data"><span class="header-section-number">1.1</span> Import all data</a></li>
  </ul></li>
  <li><a href="#subset-only-microcolonizer-data" id="toc-subset-only-microcolonizer-data" class="nav-link" data-scroll-target="#subset-only-microcolonizer-data"><span class="header-section-number">2</span> Subset only Microcolonizer data</a>
  <ul class="collapse">
  <li><a href="#plot-temperature-profiles-only" id="toc-plot-temperature-profiles-only" class="nav-link" data-scroll-target="#plot-temperature-profiles-only"><span class="header-section-number">2.1</span> Plot temperature profiles only</a></li>
  <li><a href="#join-temperature-and-tax-information" id="toc-join-temperature-and-tax-information" class="nav-link" data-scroll-target="#join-temperature-and-tax-information"><span class="header-section-number">2.2</span> Join temperature and tax information</a></li>
  <li><a href="#plot-species-richness-total-asvs-with-temperature" id="toc-plot-species-richness-total-asvs-with-temperature" class="nav-link" data-scroll-target="#plot-species-richness-total-asvs-with-temperature"><span class="header-section-number">2.3</span> Plot species richness (total ASVs) with temperature</a></li>
  </ul></li>
  <li><a href="#paring-down-the-data" id="toc-paring-down-the-data" class="nav-link" data-scroll-target="#paring-down-the-data"><span class="header-section-number">3</span> Paring down the data</a>
  <ul class="collapse">
  <li><a href="#eukaryotes" id="toc-eukaryotes" class="nav-link" data-scroll-target="#eukaryotes"><span class="header-section-number">3.1</span> Eukaryotes</a></li>
  <li><a href="#bacteria" id="toc-bacteria" class="nav-link" data-scroll-target="#bacteria"><span class="header-section-number">3.2</span> Bacteria</a></li>
  </ul></li>
  <li><a href="#get-data-imported-for-phyloseq" id="toc-get-data-imported-for-phyloseq" class="nav-link" data-scroll-target="#get-data-imported-for-phyloseq"><span class="header-section-number">4</span> Get data imported for phyloseq</a></li>
  <li><a href="#run-speic-easi" id="toc-run-speic-easi" class="nav-link" data-scroll-target="#run-speic-easi"><span class="header-section-number">5</span> Run SPEIC-EASI</a></li>
  <li><a href="#network-analysis" id="toc-network-analysis" class="nav-link" data-scroll-target="#network-analysis"><span class="header-section-number">6</span> Network analysis</a>
  <ul class="collapse">
  <li><a href="#spearman-rank-correlation" id="toc-spearman-rank-correlation" class="nav-link" data-scroll-target="#spearman-rank-correlation"><span class="header-section-number">6.1</span> Spearman Rank correlation</a></li>
  <li><a href="#other-networks" id="toc-other-networks" class="nav-link" data-scroll-target="#other-networks"><span class="header-section-number">6.2</span> Other networks?</a></li>
  </ul></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info"><span class="header-section-number">7</span> Session Info</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">microcolonizer analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="set-up-r-environment" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Set up R environment</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggplot2' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tidyr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'readr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'dplyr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'stringr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="import-all-data" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="import-all-data"><span class="header-section-number">1.1</span> Import all data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/GR_Microcolonizer_data.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  asv16s_df
  asv_wtax_18</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/temp-data.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  log_files_all</code></pre>
</div>
</div>
<p><code>asv_wtax_18</code> data includes 18S rRNA gene sequences from hydrothermal vents at Gorda Ridge and the microcolonizer substrates.</p>
<p><code>asv16s_df</code> same sample set as 18S, but with 16S rRNA gene sequences.</p>
</section>
</section>
<section id="subset-only-microcolonizer-data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Subset only Microcolonizer data</h1>
<p><em>GOALS</em></p>
<ol type="1">
<li><p>Visualize a summary of the protistan (18S rRNA gene) and bacteria/archaeal (16S rRNA gene) community within diffuse vents, at Mt Edwards especially, and then an overall look at the different microcolonizers.</p></li>
<li><p>Look at different microcolonizer samples with respect to the temperature profile. Are the communities different by temperature.</p></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(asv_wtax_18)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 29
  SAMPLE     FeatureID SEQUENCE_COUNT Taxon Domain Supergroup Phylum Class Order
  &lt;chr&gt;      &lt;chr&gt;              &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;
1 GordaRidg… 00056209…              8 Euka… Eukar… Stramenop… Sagen… &lt;NA&gt;  &lt;NA&gt; 
2 GordaRidg… 00056209…             13 Euka… Eukar… Stramenop… Sagen… &lt;NA&gt;  &lt;NA&gt; 
3 GordaRidg… 00096455…             91 Euka… Eukar… Rhizaria   Radio… Acan… &lt;NA&gt; 
4 GordaRidg… 00165708…              1 Euka… Eukar… Stramenop… Ochro… Pela… Pela…
5 GordaRidg… 0030ad8c…             36 Euka… Eukar… Stramenop… Opalo… MAST… MAST…
6 GordaRidg… 0030ad8c…             12 Euka… Eukar… Stramenop… Opalo… MAST… MAST…
# ℹ 20 more variables: Family &lt;chr&gt;, Genus &lt;chr&gt;, Species &lt;chr&gt;, VENT &lt;chr&gt;,
#   COORDINATES &lt;chr&gt;, SITE &lt;chr&gt;, SAMPLEID &lt;chr&gt;, SAMPLETYPE &lt;chr&gt;,
#   DEPTH &lt;chr&gt;, temp &lt;dbl&gt;, pH &lt;dbl&gt;, percseawater &lt;dbl&gt;, mg &lt;dbl&gt;, h2 &lt;chr&gt;,
#   h2s &lt;dbl&gt;, ch4 &lt;dbl&gt;, ProkConc &lt;chr&gt;, MC &lt;chr&gt;, dataset &lt;chr&gt;,
#   Substrate &lt;chr&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>mc_18s <span class="ot">&lt;-</span> asv_wtax_18 <span class="sc">|&gt;</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SAMPLETYPE <span class="sc">==</span> <span class="st">"Microcolonizer"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(MC, Substrate, FeatureID, Taxon, Domain, </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>           Supergroup, Phylum, Class, Order, Family, Genus, Species) <span class="sc">|&gt;</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SEQ_AVG =</span> <span class="fu">mean</span>(SEQUENCE_COUNT)) <span class="sc">|&gt;</span> </span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(TAX_LEVEL, Supergroup, Phylum, <span class="at">sep =</span> <span class="st">"-"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'MC', 'Substrate', 'FeatureID', 'Taxon',
'Domain', 'Supergroup', 'Phylum', 'Class', 'Order', 'Family', 'Genus'. You can
override using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_18s)</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># head(asv16s_df)</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>mc_16s <span class="ot">&lt;-</span> asv16s_df <span class="sc">|&gt;</span> </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sampletype <span class="sc">==</span> <span class="st">"Microcolonizer"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(MC, Substrate, FeatureID, Taxon, Domain, </span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>          Phylum, Class, Order, Family, Genus, Species) <span class="sc">|&gt;</span> </span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SEQ_AVG =</span> <span class="fu">mean</span>(SEQUENCE_COUNT)) <span class="sc">|&gt;</span> </span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(TAX_LEVEL, Domain, Phylum, <span class="at">sep =</span> <span class="st">"-"</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'MC', 'Substrate', 'FeatureID', 'Taxon',
'Domain', 'Phylum', 'Class', 'Order', 'Family', 'Genus'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_16s)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="plot-temperature-profiles-only" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="plot-temperature-profiles-only"><span class="header-section-number">2.1</span> Plot temperature profiles only</h2>
<p>Get temperature</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(log_files_all)</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Factor by colors and pairs of MCs</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>mc_ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">6</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>log_files_all<span class="sc">$</span>MC_ORDER <span class="ot">&lt;-</span> <span class="fu">factor</span>(log_files_all<span class="sc">$</span>MC, <span class="at">levels =</span> mc_ids)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>mc_col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#d7301f"</span>, <span class="st">"#4a1486"</span>, <span class="st">"#9e9ac8"</span>, <span class="st">"#fc8d59"</span>, <span class="st">"#2171b5"</span>, <span class="st">"#6baed6"</span>)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(mc_col) <span class="ot">&lt;-</span> mc_ids</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>temps <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(log_files_all, <span class="fu">aes</span>(<span class="at">x =</span> Parsed_time_EST, <span class="at">y =</span> Value, <span class="at">color =</span> <span class="fu">as.factor</span>(MC_ORDER))) <span class="sc">+</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_step</span>() <span class="sc">+</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mc_col) <span class="sc">+</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Temperature •C"</span>) <span class="sc">+</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>temps</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="analysis-microcolonizers_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="join-temperature-and-tax-information" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="join-temperature-and-tax-information"><span class="header-section-number">2.2</span> Join temperature and tax information</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>temp_row <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(log_files_all, <span class="fu">aes</span>(<span class="at">x =</span> Parsed_time_EST, <span class="at">y =</span> Value, <span class="at">color =</span> <span class="fu">as.factor</span>(MC_ORDER))) <span class="sc">+</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_step</span>() <span class="sc">+</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mc_col) <span class="sc">+</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(MC_ORDER)) <span class="sc">+</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Temperature •C"</span>) <span class="sc">+</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"left"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_16s)</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>bac_row <span class="ot">&lt;-</span> mc_16s <span class="sc">|&gt;</span> </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(TAX_LEVEL, MC, Substrate) <span class="sc">|&gt;</span> </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SEQ_AVG)) <span class="sc">|&gt;</span> </span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Substrate, <span class="at">y =</span> SUM, <span class="at">fill =</span> TAX_LEVEL)) <span class="sc">+</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(MC)) <span class="sc">+</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'TAX_LEVEL', 'MC'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>bac_row</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="analysis-microcolonizers_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_16s)</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>euk_row <span class="ot">&lt;-</span> mc_18s <span class="sc">|&gt;</span> </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(TAX_LEVEL, MC, Substrate) <span class="sc">|&gt;</span> </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SEQ_AVG)) <span class="sc">|&gt;</span> </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Substrate, <span class="at">y =</span> SUM, <span class="at">fill =</span> TAX_LEVEL)) <span class="sc">+</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(MC)) <span class="sc">+</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'TAX_LEVEL', 'MC'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>euk_row</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="analysis-microcolonizers_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>temp_row <span class="sc">+</span> bac_row <span class="sc">+</span> euk_row <span class="sc">+</span> patchwork<span class="sc">::</span><span class="fu">plot_layout</span>(<span class="at">nrow =</span> <span class="dv">1</span>,<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.8</span>, <span class="fl">0.4</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="analysis-microcolonizers_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</section>
<section id="plot-species-richness-total-asvs-with-temperature" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="plot-species-richness-total-asvs-with-temperature"><span class="header-section-number">2.3</span> Plot species richness (total ASVs) with temperature</h2>
<p><em>Question</em>: Does overall species richness change within the microcolonizers? over “time” and with respect to temperature. Same for protists and prokaryotes?</p>
<p>Show differences between microcolonizers with temperature.</p>
<p><em>continuing question</em>: if there are specific patterns in species richness for certain microcolonizers, are there certain species that are driving this pattern?</p>
</section>
</section>
<section id="paring-down-the-data" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Paring down the data</h1>
<section id="eukaryotes" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="eukaryotes"><span class="header-section-number">3.1</span> Eukaryotes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(asv_wtax_18)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 29
  SAMPLE     FeatureID SEQUENCE_COUNT Taxon Domain Supergroup Phylum Class Order
  &lt;chr&gt;      &lt;chr&gt;              &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;
1 GordaRidg… 00056209…              8 Euka… Eukar… Stramenop… Sagen… &lt;NA&gt;  &lt;NA&gt; 
2 GordaRidg… 00056209…             13 Euka… Eukar… Stramenop… Sagen… &lt;NA&gt;  &lt;NA&gt; 
3 GordaRidg… 00096455…             91 Euka… Eukar… Rhizaria   Radio… Acan… &lt;NA&gt; 
4 GordaRidg… 00165708…              1 Euka… Eukar… Stramenop… Ochro… Pela… Pela…
5 GordaRidg… 0030ad8c…             36 Euka… Eukar… Stramenop… Opalo… MAST… MAST…
6 GordaRidg… 0030ad8c…             12 Euka… Eukar… Stramenop… Opalo… MAST… MAST…
# ℹ 20 more variables: Family &lt;chr&gt;, Genus &lt;chr&gt;, Species &lt;chr&gt;, VENT &lt;chr&gt;,
#   COORDINATES &lt;chr&gt;, SITE &lt;chr&gt;, SAMPLEID &lt;chr&gt;, SAMPLETYPE &lt;chr&gt;,
#   DEPTH &lt;chr&gt;, temp &lt;dbl&gt;, pH &lt;dbl&gt;, percseawater &lt;dbl&gt;, mg &lt;dbl&gt;, h2 &lt;chr&gt;,
#   h2s &lt;dbl&gt;, ch4 &lt;dbl&gt;, ProkConc &lt;chr&gt;, MC &lt;chr&gt;, dataset &lt;chr&gt;,
#   Substrate &lt;chr&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(asv_wtax_18<span class="sc">$</span>Domain)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Eukaryota"  "Bacteria"   "Unassigned"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>mc_tmp_18 <span class="ot">&lt;-</span> asv_wtax_18 <span class="sc">|&gt;</span> </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>SAMPLE) <span class="sc">|&gt;</span> </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SAMPLETYPE <span class="sc">==</span> <span class="st">"Microcolonizer"</span> <span class="sc">|</span> VENT <span class="sc">==</span> <span class="st">"Mt Edwards"</span> <span class="sc">|</span> VENT <span class="sc">==</span> <span class="st">"Deep seawater"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUBSTRATE =</span> <span class="fu">str_replace_all</span>(Substrate, <span class="st">"z2"</span>, <span class="st">"z"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE =</span> <span class="fu">case_when</span>(</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>    SAMPLETYPE <span class="sc">==</span> <span class="st">"Microcolonizer"</span> <span class="sc">~</span> <span class="fu">paste</span>(MC, SUBSTRATE, <span class="at">sep =</span> <span class="st">"-"</span>),</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>    VENT <span class="sc">==</span> <span class="st">"Deep seawater"</span> <span class="sc">~</span> <span class="st">"Background seawater"</span>,</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>    VENT <span class="sc">==</span> <span class="st">"Mt Edwards"</span> <span class="sc">~</span> <span class="st">"Mt Edwards diffuse fluid"</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">|&gt;</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, FeatureID, Taxon, Domain, </span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>           Supergroup, Phylum, Class, Order, Family, Genus, Species) <span class="sc">|&gt;</span> </span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SEQ_AVG =</span> <span class="fu">mean</span>(SEQUENCE_COUNT)) <span class="sc">|&gt;</span> </span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SEQ_AVG <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE', 'FeatureID', 'Taxon', 'Domain',
'Supergroup', 'Phylum', 'Class', 'Order', 'Family', 'Genus'. You can override
using the `.groups` argument.</code></pre>
</div>
</div>
<p>Filter by ASV size (sequence total for ASVs)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ASV filtering by how many sequences are present for each ASV</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Total number of ASVs</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(mc_tmp_18<span class="sc">$</span>FeatureID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2168</code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Total number of ASVs that have at least 500 sequences in one of the samples.</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>tmp2 <span class="ot">&lt;-</span> mc_tmp_18 <span class="sc">|&gt;</span> </span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SEQ_AVG <span class="sc">&gt;</span> <span class="dv">100</span>)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(tmp2<span class="sc">$</span>FeatureID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 450</code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>keep_18 <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(tmp2<span class="sc">$</span>FeatureID))</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(keep_18)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 450</code></pre>
</div>
</div>
<p>Create filtered 18S dataset and convert to phyloseq object.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mc_tmp_18)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 12
# Groups:   SAMPLE, FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order,
#   Family, Genus [6]
  SAMPLE       FeatureID Taxon Domain Supergroup Phylum Class Order Family Genus
  &lt;chr&gt;        &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;
1 Background … 00165708… Euka… Eukar… Stramenop… Ochro… Pela… Pela… Pelag… Pela…
2 Background … 0038478b… Euka… Eukar… Opisthoko… Metaz… Cnid… Cnid… Hydro… &lt;NA&gt; 
3 Background … 0132268c… Euka… Eukar… Alveolata  Cilio… Spir… &lt;NA&gt;  &lt;NA&gt;   &lt;NA&gt; 
4 Background … 01a71ee7… Euka… Eukar… Stramenop… Pseud… MAST… MAST… MAST-… MAST…
5 Background … 01d1a4a1… Euka… Eukar… Rhizaria   Radio… RAD-B RAD-… RAD-B… RAD-…
6 Background … 02029510… Euka… Eukar… Rhizaria   Radio… RAD-B RAD-… RAD-B… RAD-…
# ℹ 2 more variables: Species &lt;chr&gt;, SEQ_AVG &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>mat_18s_filtered <span class="ot">&lt;-</span> mc_tmp_18 <span class="sc">|&gt;</span> <span class="fu">ungroup</span>() <span class="sc">|&gt;</span> </span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FeatureID <span class="sc">%in%</span> keep_18) <span class="sc">|&gt;</span> </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SEQ_AVG <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> </span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FeatureID, SAMPLE, SEQ_AVG) <span class="sc">|&gt;</span> </span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> SAMPLE, <span class="at">values_from =</span> SEQ_AVG, <span class="at">values_fill =</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> </span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">MC4-Shell</span><span class="st">`</span>) <span class="sc">|&gt;</span> </span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"FeatureID"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(mat_18s_filtered)</span></span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>mat_tax18_filtered <span class="ot">&lt;-</span> mc_tmp_18 <span class="sc">|&gt;</span> <span class="fu">ungroup</span>() <span class="sc">|&gt;</span> </span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FeatureID <span class="sc">%in%</span> keep_18) <span class="sc">|&gt;</span> </span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FeatureID, Taxon, Domain, Phylum, Class, Order, Family, Genus, Species) <span class="sc">|&gt;</span> </span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">|&gt;</span> </span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"FeatureID"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>SPEIC-EASI network analysis requires samples to be in the same order. Additionally, the bacteria set of samples has more substrate options than the eukaryote dataset.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mat_18s_filtered)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Background seawater"      "MC1-Quartz"              
 [3] "MC1-Riftia"               "MC1-Shell"               
 [5] "MC2-Quartz"               "MC2-Riftia"              
 [7] "MC3-Riftia"               "MC3-Shell"               
 [9] "MC4-Quartz"               "MC5-Quartz"              
[11] "MC5-Riftia"               "MC5-Shell"               
[13] "MC6-Quartz"               "MC6-Riftia"              
[15] "MC6-Shell"                "Mt Edwards diffuse fluid"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>set_of_samples <span class="ot">&lt;-</span> <span class="fu">colnames</span>(mat_18s_filtered) <span class="co"># Make character list</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="co"># class(set_of_samples)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bacteria" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="bacteria"><span class="header-section-number">3.2</span> Bacteria</h2>
<p>As long as an ASV appears at least once to have more than 100 sequences, keep the ASV.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(asv16s_df)</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>mc_tmp <span class="ot">&lt;-</span> asv16s_df <span class="sc">|&gt;</span> </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sampletype <span class="sc">==</span> <span class="st">"Microcolonizer"</span> <span class="sc">|</span> LocationName <span class="sc">==</span> <span class="st">"Mt Edwards Vent"</span> <span class="sc">|</span> LocationName <span class="sc">==</span> <span class="st">"Deep seawater"</span> <span class="sc">&amp;</span> STATUS <span class="sc">==</span> <span class="st">"keep"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE =</span> <span class="fu">case_when</span>(</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>    Sampletype <span class="sc">==</span> <span class="st">"Microcolonizer"</span> <span class="sc">~</span> <span class="fu">paste</span>(MC, Substrate, <span class="at">sep =</span> <span class="st">"-"</span>),</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>    LocationName <span class="sc">==</span> <span class="st">"Deep seawater"</span> <span class="sc">~</span> <span class="st">"Background seawater"</span>,</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>    LocationName <span class="sc">==</span> <span class="st">"Mt Edwards Vent"</span> <span class="sc">~</span> <span class="st">"Mt Edwards diffuse fluid"</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">|&gt;</span> </span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, FeatureID, Taxon, Domain, </span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>          Phylum, Class, Order, Family, Genus, Species) <span class="sc">|&gt;</span> </span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SEQ_AVG =</span> <span class="fu">mean</span>(SEQUENCE_COUNT)) <span class="sc">|&gt;</span> </span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(Domain <span class="sc">==</span> <span class="st">"Unassigned"</span>) <span class="sc">&amp;</span> <span class="sc">!</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SEQ_AVG <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE', 'FeatureID', 'Taxon', 'Domain',
'Phylum', 'Class', 'Order', 'Family', 'Genus'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ASV filtering by how many sequences are present for each ASV</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Total number of ASVs</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(mc_tmp<span class="sc">$</span>FeatureID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5129</code></pre>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Total number of ASVs that have at least 100 sequences in one of the samples.</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>tmptmp <span class="ot">&lt;-</span> mc_tmp <span class="sc">|&gt;</span> </span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SEQ_AVG <span class="sc">&gt;</span> <span class="dv">100</span>)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(tmptmp<span class="sc">$</span>FeatureID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1331</code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(tmptmp<span class="sc">$</span>FeatureID))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1331</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>set_of_samples</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Background seawater"      "MC1-Quartz"              
 [3] "MC1-Riftia"               "MC1-Shell"               
 [5] "MC2-Quartz"               "MC2-Riftia"              
 [7] "MC3-Riftia"               "MC3-Shell"               
 [9] "MC4-Quartz"               "MC5-Quartz"              
[11] "MC5-Riftia"               "MC5-Shell"               
[13] "MC6-Quartz"               "MC6-Riftia"              
[15] "MC6-Shell"                "Mt Edwards diffuse fluid"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tmptmp<span class="sc">$</span>SAMPLE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Background seawater"      "MC1-Olivine"             
 [3] "MC1-Quartz"               "MC1-Riftia"              
 [5] "MC1-Shell"                "MC2-Basalt"              
 [7] "MC2-Olivine"              "MC2-Quartz"              
 [9] "MC2-Riftia"               "MC3-Basalt"              
[11] "MC3-Olivine"              "MC3-Quartz"              
[13] "MC3-Riftia"               "MC3-Shell"               
[15] "MC4-Basalt"               "MC4-Pyrite"              
[17] "MC4-Quartz"               "MC4-Riftia"              
[19] "MC5-Basalt"               "MC5-Pyrite"              
[21] "MC5-Quartz"               "MC5-Riftia"              
[23] "MC5-Shell"                "MC6-Basalt"              
[25] "MC6-Pyrite"               "MC6-Quartz"              
[27] "MC6-Riftia"               "MC6-Shell"               
[29] "Mt Edwards diffuse fluid"</code></pre>
</div>
</div>
<p>Create filtered 16S dataset and convert to phyloseq object.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(mc_tmp)</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>mat_16s_filtered <span class="ot">&lt;-</span> mc_tmp <span class="sc">|&gt;</span> <span class="fu">ungroup</span>() <span class="sc">|&gt;</span> </span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FeatureID <span class="sc">%in%</span> keep) <span class="sc">|&gt;</span> </span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SEQ_AVG <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> </span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FeatureID, SAMPLE, SEQ_AVG) <span class="sc">|&gt;</span> </span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> SAMPLE, <span class="at">values_from =</span> SEQ_AVG, <span class="at">values_fill =</span> <span class="dv">0</span>) <span class="sc">|&gt;</span> </span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">|&gt;</span> </span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FeatureID, <span class="fu">all_of</span>(set_of_samples)) <span class="sc">|&gt;</span> </span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"FeatureID"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mat_16s_filtered)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Background seawater"      "MC1-Quartz"              
 [3] "MC1-Riftia"               "MC1-Shell"               
 [5] "MC2-Quartz"               "MC2-Riftia"              
 [7] "MC3-Riftia"               "MC3-Shell"               
 [9] "MC4-Quartz"               "MC5-Quartz"              
[11] "MC5-Riftia"               "MC5-Shell"               
[13] "MC6-Quartz"               "MC6-Riftia"              
[15] "MC6-Shell"                "Mt Edwards diffuse fluid"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>mat_tax_filtered <span class="ot">&lt;-</span> mc_tmp <span class="sc">|&gt;</span> <span class="fu">ungroup</span>() <span class="sc">|&gt;</span> </span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FeatureID <span class="sc">%in%</span> keep) <span class="sc">|&gt;</span> </span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FeatureID, Taxon, Domain, Phylum, Class, Order, Family, Genus, Species) <span class="sc">|&gt;</span> </span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">|&gt;</span> </span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"FeatureID"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">colnames</span>(mat_16s_filtered), <span class="fu">colnames</span>(mat_18s_filtered))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
</section>
</section>
<section id="get-data-imported-for-phyloseq" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Get data imported for phyloseq</h1>
<p>The goal is to run a network analysis Use <code>mat_tax_filtered</code> and <code>mat_16s_filtered</code> for bacteria and archaea members of the community.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>asv_16s <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(mat_16s_filtered, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>tax_16s <span class="ot">&lt;-</span> <span class="fu">tax_table</span>(mat_tax_filtered)</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>phylo_16s <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(asv_16s, tax_16s)</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>phylo_16s</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1331 taxa and 16 samples ]
tax_table()   Taxonomy Table:    [ 1331 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</div>
<p>Use <code>mat_tax18_filtered</code> and <code>mat_18s_filtered</code> for eukaryote members of the community.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>asv_18s <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(mat_18s_filtered, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>tax_18s <span class="ot">&lt;-</span> <span class="fu">tax_table</span>(mat_tax18_filtered)</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>phylo_18s <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(asv_18s, tax_18s)</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>phylo_18s</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 450 taxa and 16 samples ]
tax_table()   Taxonomy Table:    [ 450 taxa by 8 taxonomic ranks ]</code></pre>
</div>
</div>
<p>Both should have 16 samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(phylo_16s, phylo_18s, <span class="at">file =</span> <span class="st">"output-data/phyloseq-objs.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-speic-easi" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Run SPEIC-EASI</h1>
<p>Installation of SpiecEasi and phyloseq on the HPRC</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: false</span></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="co"># library(devtools)</span></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a><span class="co"># install_github("zdk123/SpiecEasi")</span></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="co"># if (!require("BiocManager", quietly = TRUE))</span></span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># install.packages("BiocManager")</span></span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("phyloseq")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>For network analysis, we will take filtered 18S and 16S rRNA gene dataset and perform the network analysis across the two.</p>
<p>For the substrate at Gorda Ridge, I’ve also included the background seawater and the diffuse vent fluid from Mt. Edwards.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: false</span></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="co"># load("/16s-microcolonizer/network/phyloseq-objs.RData", verbose = TRUE)</span></span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="co"># phylo_16s</span></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a><span class="co"># phylo_18s</span></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a><span class="co"># library(phyloseq); library(SpiecEasi); library(tidyverse)</span></span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a><span class="co"># ?spiec.easi Cross Domain approach </span></span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a><span class="co"># substrate_GR &lt;- spiec.easi(list(phylo_16s, phylo_18s), </span></span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a><span class="co">#                     method = 'mb', verbose = TRUE,</span></span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                     nlambda = 40, lambda.min.ratio = 1e-2, </span></span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a><span class="co">#                     pulsar.params = list(thresh = 0.05),</span></span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                     ncores = 8)</span></span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a><span class="co"># save(substrate_GR, file = "/scratch/group/hu-lab/microcolonizer-gr-2019/substrate-gr-network.RData")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li><code>mb</code>: meinshausen-buhlmann’s neighborhood selection</li>
</ul>
<p>Reference from https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-017-0393-0</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Check output</span></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a><span class="co"># getStability(se_GR)</span></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="co"># sum(getRefit(se_GR))/2</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a><span class="co"># # Extract weighted matrix </span></span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a><span class="co"># se.beta &lt;- as.matrix(symBeta(getOptBeta(se_GR)))</span></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a><span class="co"># df_beta &lt;- as.data.frame(se.beta)</span></span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Extract adajency matrix </span></span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a><span class="co"># adj.mat &lt;- getRefit(se_GR)</span></span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a><span class="co"># df_adj &lt;- as.data.frame(as.matrix(adj.mat))</span></span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Assign names from original dataframes </span></span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(df_beta) &lt;- colnames(se_GR$est$data) </span></span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(df_adj) &lt;- colnames(se_GR$est$data)</span></span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a><span class="co"># row.names(df_adj) &lt;- colnames(se_GR$est$data)</span></span>
<span id="cb71-19"><a href="#cb71-19" aria-hidden="true" tabindex="-1"></a><span class="co"># row.names(df_beta) &lt;- colnames(se_GR$est$data)</span></span>
<span id="cb71-20"><a href="#cb71-20" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb71-21"><a href="#cb71-21" aria-hidden="true" tabindex="-1"></a><span class="co"># # Save output </span></span>
<span id="cb71-22"><a href="#cb71-22" aria-hidden="true" tabindex="-1"></a><span class="co"># save(df_adj, df_beta, se_GR, file = 'gr-spieceasi-output-20-08-2020.RData')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="network-analysis" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Network analysis</h1>
<section id="spearman-rank-correlation" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="spearman-rank-correlation"><span class="header-section-number">6.1</span> Spearman Rank correlation</h2>
<p>Use the same pared down data to perform a spearman correlation coefficient test.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(compositions)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'compositions' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to compositions, a package for compositional data analysis.
Find an intro with "? compositions"</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'compositions'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    anova, cor, cov, dist, var</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:graphics':

    segments</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    %*%, norm, scale, scale.default</code></pre>
</div>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co"># CLR transform datasets individually and then anneal together.</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>combined_ASVs_clr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(compositions<span class="sc">::</span><span class="fu">clr</span>(<span class="fu">t</span>(<span class="fu">as.data.frame</span>(mat_16s_filtered))))) <span class="sc">%&gt;%</span> </span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"TMP"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TMP_2 =</span> <span class="fu">paste</span>(TMP, <span class="st">"16S"</span>, <span class="at">sep =</span> <span class="st">"_"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>    (<span class="fu">as.data.frame</span>(<span class="fu">t</span>(compositions<span class="sc">::</span><span class="fu">clr</span>(<span class="fu">t</span>(<span class="fu">as.data.frame</span>(mat_18s_filtered))))) <span class="sc">%&gt;%</span> </span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"TMP"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TMP_2 =</span> <span class="fu">paste</span>(TMP, <span class="st">"18S"</span>, <span class="at">sep =</span> <span class="st">"_"</span>)))</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"TMP_2"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>TMP)</span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(combined_ASVs_clr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 1,781
Columns: 16
$ `Background seawater`      &lt;dbl&gt; -1.2025536, -3.1484637, 0.3349718, -3.84161…
$ `MC1-Quartz`               &lt;dbl&gt; 0.0000000, 0.0000000, 0.9688627, 0.0000000,…
$ `MC1-Riftia`               &lt;dbl&gt; 0.0000000, 0.0000000, 0.0000000, 0.0000000,…
$ `MC1-Shell`                &lt;dbl&gt; 0.00000000, 0.00000000, 0.00000000, 0.00000…
$ `MC2-Quartz`               &lt;dbl&gt; 0.0000000, 0.0000000, 0.2370333, 0.5658913,…
$ `MC2-Riftia`               &lt;dbl&gt; 0.000000, 0.000000, 0.000000, 0.000000, 0.0…
$ `MC3-Riftia`               &lt;dbl&gt; 0.000000, 0.000000, 0.000000, 0.000000, 0.0…
$ `MC3-Shell`                &lt;dbl&gt; 0.0000000, 0.0000000, 0.0000000, 0.0000000,…
$ `MC4-Quartz`               &lt;dbl&gt; 0.0000000, 0.0000000, -0.8885662, 0.0000000…
$ `MC5-Quartz`               &lt;dbl&gt; 0.0000000, 0.0000000, 0.7677860, 0.0000000,…
$ `MC5-Riftia`               &lt;dbl&gt; 0.000000, 0.000000, 0.000000, 0.000000, 0.0…
$ `MC5-Shell`                &lt;dbl&gt; 0.000000, 0.000000, 0.000000, 0.000000, 0.0…
$ `MC6-Quartz`               &lt;dbl&gt; 0.000000, 0.000000, 0.000000, 0.000000, 0.0…
$ `MC6-Riftia`               &lt;dbl&gt; 0.000000, 0.000000, 0.000000, 0.000000, 0.0…
$ `MC6-Shell`                &lt;dbl&gt; 0.000000, 0.000000, 0.000000, 0.000000, 0.0…
$ `Mt Edwards diffuse fluid` &lt;dbl&gt; 0.0000000, 0.0000000, -0.1616575, 0.0000000…</code></pre>
</div>
</div>
<p>Calculate spearman correlation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ?cor()</span></span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>spear_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(<span class="fu">t</span>(combined_ASVs_clr), <span class="at">method =</span> <span class="st">"spearman"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in stats::cor(x, y, use, method): the standard deviation is zero</code></pre>
</div>
</div>
<p><code>spear_cor</code> is a matrix that has all pairwise comparisons of ASVs to one another.</p>
<p>Spearman’s rank correlation coefficient shows the ranking between two variables (all pairwise comparisons). Range of values is from -1 to 1. Values that are closer to -1 or 1 are the most opposed (-1) or most identical (+1).</p>
<p>For our analysis, we will look at comparisons that are &lt;-0.5 for negative relationships and and &gt;0.5 for positive relationships.</p>
<p>Re-format data so it is meaningful</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>spear_cor_output <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(spear_cor) <span class="sc">%&gt;%</span> </span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"PAIR_A"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="sc">-</span>PAIR_A), <span class="at">names_to =</span> <span class="st">"PAIR_B"</span>, <span class="at">values_to =</span> <span class="st">"VALUE"</span>)</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a><span class="co"># head(spear_cor_output)</span></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(spear_cor_output<span class="sc">$</span>VALUE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="analysis-microcolonizers_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Stats on Spearman output</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(spear_cor_output)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 3,171,961
Columns: 3
$ PAIR_A &lt;chr&gt; "04ecfad5772d2e09a84a0f5ef460536c_16S", "04ecfad5772d2e09a84a0f…
$ PAIR_B &lt;chr&gt; "04ecfad5772d2e09a84a0f5ef460536c_16S", "0562e97bf8665edd5ae858…
$ VALUE  &lt;dbl&gt; 1.0000000, 1.0000000, -0.3539879, 0.7302967, -0.7302967, 0.4232…</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>filtered_spear_cor <span class="ot">&lt;-</span> spear_cor_output <span class="sc">%&gt;%</span> </span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>((VALUE <span class="sc">&gt;</span> <span class="fl">0.75</span> <span class="sc">|</span> VALUE <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.75</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(PAIR_A <span class="sc">==</span> PAIR_B))</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(filtered_spear_cor)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 18,622
Columns: 3
$ PAIR_A &lt;chr&gt; "04ecfad5772d2e09a84a0f5ef460536c_16S", "04ecfad5772d2e09a84a0f…
$ PAIR_B &lt;chr&gt; "0562e97bf8665edd5ae85899297a44a8_16S", "09de3dde6c465b74bcb8e6…
$ VALUE  &lt;dbl&gt; 1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 1, -1, -1, -1, 1…</code></pre>
</div>
</div>
<ul>
<li><p>We began with over 3 million comparisons.</p></li>
<li><p>We filtered by abs(0.75) and we are ending up with over 20,000 pairwise interactions to look at.</p></li>
<li><p>Also removed pairwise comparisons that are identical to one another.</p></li>
<li><p>Ending with over 18k comparisons.</p></li>
</ul>
<p>Include taxonomic information</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>taxa_key_combined <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mat_tax_filtered) <span class="sc">%&gt;%</span> </span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"bac"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FEATUREID =</span> <span class="fu">paste</span>(bac, <span class="st">"16S"</span>, <span class="at">sep =</span> <span class="st">"_"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>((<span class="fu">as.data.frame</span>(mat_tax18_filtered) <span class="sc">%&gt;%</span> </span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"bac"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">FEATUREID =</span> <span class="fu">paste</span>(bac, <span class="st">"18S"</span>, <span class="at">sep =</span> <span class="st">"_"</span>)))) <span class="sc">%&gt;%</span> </span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>bac)</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a><span class="co"># head(taxa_key_combined)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Join with taxa information in step-wise fashion.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(filtered_spear_cor)</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>filtered_spear_cor_wtax <span class="ot">&lt;-</span> filtered_spear_cor <span class="sc">%&gt;%</span> </span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(taxa_key_combined, <span class="at">by =</span> <span class="fu">join_by</span>(PAIR_A <span class="sc">==</span> FEATUREID)) <span class="sc">%&gt;%</span> </span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(taxa_key_combined, <span class="at">by =</span> <span class="fu">join_by</span>(PAIR_B <span class="sc">==</span> FEATUREID), <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">"_A"</span>, <span class="st">"_B"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Adding interaction type at the domain level</span></span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">INTERACTION_TYPE =</span> <span class="fu">paste</span>(Domain_A, Domain_B, <span class="at">sep =</span> <span class="st">"-"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filtering again if phylum level is NA</span></span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(<span class="fu">is.na</span>(Phylum_A)) <span class="sc">|</span> <span class="sc">!</span>(<span class="fu">is.na</span>(Phylum_B)))</span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(filtered_spear_cor_wtax)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 20
  PAIR_A PAIR_B VALUE Taxon_A Domain_A Phylum_A Class_A Order_A Family_A Genus_A
  &lt;chr&gt;  &lt;chr&gt;  &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;  
1 04ecf… 0562e…     1 d__Bac… Bacteria Proteob… Alphap… Rhizob… Beijeri… Methyl…
2 04ecf… 09de3…    -1 d__Bac… Bacteria Proteob… Alphap… Rhizob… Beijeri… Methyl…
3 04ecf… 0f118…    -1 d__Bac… Bacteria Proteob… Alphap… Rhizob… Beijeri… Methyl…
4 04ecf… 1040e…    -1 d__Bac… Bacteria Proteob… Alphap… Rhizob… Beijeri… Methyl…
5 04ecf… 12696…    -1 d__Bac… Bacteria Proteob… Alphap… Rhizob… Beijeri… Methyl…
6 04ecf… 12c4c…    -1 d__Bac… Bacteria Proteob… Alphap… Rhizob… Beijeri… Methyl…
# ℹ 10 more variables: Species_A &lt;chr&gt;, Taxon_B &lt;chr&gt;, Domain_B &lt;chr&gt;,
#   Phylum_B &lt;chr&gt;, Class_B &lt;chr&gt;, Order_B &lt;chr&gt;, Family_B &lt;chr&gt;,
#   Genus_B &lt;chr&gt;, Species_B &lt;chr&gt;, INTERACTION_TYPE &lt;chr&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(filtered_spear_cor_wtax<span class="sc">$</span>INTERACTION_TYPE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Bacteria-Bacteria"   "Bacteria-Archaea"    "Bacteria-Eukaryota" 
[4] "Archaea-Archaea"     "Archaea-Bacteria"    "Archaea-Eukaryota"  
[7] "Eukaryota-Eukaryota" "Eukaryota-Archaea"   "Eukaryota-Bacteria" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(filtered_spear_cor_wtax)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 18,622
Columns: 20
$ PAIR_A           &lt;chr&gt; "04ecfad5772d2e09a84a0f5ef460536c_16S", "04ecfad5772d…
$ PAIR_B           &lt;chr&gt; "0562e97bf8665edd5ae85899297a44a8_16S", "09de3dde6c46…
$ VALUE            &lt;dbl&gt; 1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 1, -1,…
$ Taxon_A          &lt;chr&gt; "d__Bacteria; p__Proteobacteria; c__Alphaproteobacter…
$ Domain_A         &lt;chr&gt; "Bacteria", "Bacteria", "Bacteria", "Bacteria", "Bact…
$ Phylum_A         &lt;chr&gt; "Proteobacteria", "Proteobacteria", "Proteobacteria",…
$ Class_A          &lt;chr&gt; "Alphaproteobacteria", "Alphaproteobacteria", "Alphap…
$ Order_A          &lt;chr&gt; "Rhizobiales", "Rhizobiales", "Rhizobiales", "Rhizobi…
$ Family_A         &lt;chr&gt; "Beijerinckiaceae", "Beijerinckiaceae", "Beijerinckia…
$ Genus_A          &lt;chr&gt; "Methylobacterium-Methylorubrum", "Methylobacterium-M…
$ Species_A        &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ Taxon_B          &lt;chr&gt; "d__Bacteria; p__Deinococcota; c__Deinococci; o__Ther…
$ Domain_B         &lt;chr&gt; "Bacteria", "Bacteria", "Archaea", "Bacteria", "Bacte…
$ Phylum_B         &lt;chr&gt; "Deinococcota", "Actinobacteriota", "Crenarchaeota", …
$ Class_B          &lt;chr&gt; "Deinococci", "Actinobacteria", "Nitrososphaeria", "A…
$ Order_B          &lt;chr&gt; "Thermales", "Corynebacteriales", "Nitrosopumilales",…
$ Family_B         &lt;chr&gt; "Thermaceae", "Nocardiaceae", "Nitrosopumilaceae", "C…
$ Genus_B          &lt;chr&gt; "Thermus", "Rhodococcus", "Nitrosopumilaceae", NA, "C…
$ Species_B        &lt;chr&gt; NA, NA, "uncultured_marine", NA, NA, NA, NA, NA, "unc…
$ INTERACTION_TYPE &lt;chr&gt; "Bacteria-Bacteria", "Bacteria-Bacteria", "Bacteria-A…</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(filtered_spear_cor_wtax, <span class="at">file =</span> <span class="st">"output-data/spearman_output.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="other-networks" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="other-networks"><span class="header-section-number">6.2</span> Other networks?</h2>
<p>Graphical lasso network: Non-paranormal transformation applied to data using <code>huge.npn</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># library(huge)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run graphical lasso network</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set.seed(100)</span></span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a><span class="co"># npn &lt;- huge.npn(netGAM5)</span></span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a><span class="co"># lams  &lt;- pulsar::getLamPath(pulsar::getMaxCov(npn), .01, len=30)</span></span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a><span class="co"># hugeargs &lt;- list(lambda=lams, verbose=FALSE, method='glasso')</span></span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a><span class="co"># outd &lt;- pulsar::pulsar(npn, fun=huge::huge, fargs=hugeargs,rep.num=50, criterion = "stars")</span></span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a><span class="co"># opt &lt;- outd$stars</span></span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a><span class="co"># n &lt;- opt$opt.index</span></span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a><span class="co"># # Get output adjacency matrix from graphical lasso model</span></span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a><span class="co"># fit &lt;- pulsar::refit(outd)</span></span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a><span class="co"># fit &lt;- fit$refit</span></span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a><span class="co"># fit.fin &lt;- fit$stars</span></span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a><span class="co"># fit.fin &lt;- as.matrix(fit.fin)</span></span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a><span class="co"># fit.fin &lt;- as.data.frame(fit.fin)</span></span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(fit.fin) &lt;- colnames(netGAM5)</span></span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a><span class="co"># rownames(fit.fin)&lt;- colnames(netGAM5)</span></span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a><span class="co"># fit.fin &lt;- as.matrix(fit.fin)</span></span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(fit.fin)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="session-info" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Session Info</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.2 (2022-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.5.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] compositions_2.0-8   patchwork_1.3.0.9000 phyloseq_1.42.0     
 [4] lubridate_1.9.3      forcats_1.0.0        stringr_1.5.1       
 [7] dplyr_1.1.4          purrr_1.0.2          readr_2.1.5         
[10] tidyr_1.3.1          tibble_3.2.1         ggplot2_3.5.1       
[13] tidyverse_2.0.0     

loaded via a namespace (and not attached):
 [1] Biobase_2.58.0         jsonlite_1.8.9         splines_4.2.2         
 [4] foreach_1.5.2          stats4_4.2.2           tensorA_0.36.2.1      
 [7] GenomeInfoDbData_1.2.9 robustbase_0.95-1      yaml_2.3.10           
[10] pillar_1.9.0           lattice_0.22-6         glue_1.8.0            
[13] digest_0.6.37          XVector_0.38.0         colorspace_2.1-1      
[16] htmltools_0.5.8.1      Matrix_1.5-1           plyr_1.8.9            
[19] pkgconfig_2.0.3        zlibbioc_1.44.0        scales_1.3.0          
[22] tzdb_0.4.0             timechange_0.3.0       mgcv_1.9-1            
[25] farver_2.1.2           generics_0.1.3         IRanges_2.32.0        
[28] withr_3.0.1            BiocGenerics_0.44.0    cli_3.6.3             
[31] survival_3.7-0         magrittr_2.0.3         crayon_1.5.3          
[34] evaluate_1.0.1         fansi_1.0.6            nlme_3.1-160          
[37] MASS_7.3-58.1          vegan_2.6-4            tools_4.2.2           
[40] data.table_1.16.2      hms_1.1.3              lifecycle_1.0.4       
[43] Rhdf5lib_1.20.0        S4Vectors_0.36.2       munsell_0.5.1         
[46] cluster_2.1.6          Biostrings_2.66.0      ade4_1.7-22           
[49] compiler_4.2.2         GenomeInfoDb_1.34.9    rlang_1.1.4           
[52] rhdf5_2.42.1           grid_4.2.2             RCurl_1.98-1.12       
[55] iterators_1.0.14       rhdf5filters_1.10.1    biomformat_1.26.0     
[58] rstudioapi_0.16.0      htmlwidgets_1.6.4      igraph_2.0.3          
[61] labeling_0.4.3         bitops_1.0-9           rmarkdown_2.28        
[64] gtable_0.3.5           codetools_0.2-20       multtest_2.54.0       
[67] DBI_1.2.3              reshape2_1.4.4         R6_2.5.1              
[70] bayesm_3.1-6           knitr_1.48             fastmap_1.2.0         
[73] utf8_1.2.4             permute_0.9-7          ape_5.8               
[76] stringi_1.8.4          parallel_4.2.2         Rcpp_1.0.13           
[79] vctrs_0.6.5            DEoptimR_1.1-3         tidyselect_1.2.1      
[82] xfun_0.48             </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>